lib/handlers/post.js

```diff
@@ -1,153 +1,52 @@
-        resourceMetadata.isBasicContainer);
-
-    // Check if URI is already in use
-    if (resourcePath === null) {
-        ldp.releaseResourceUri(resourcePath);
-        debug("POST -- URI already exists or in use");
-
-        return next(new HttpError({
-            status: 400,
-            message: "URI already exists or in use"
-        }));
-    }
-
-    // Creating a graph and add the req text
-    var resourceGraph = $rdf.graph();
-    var requestText = req.convertedText || req.text;
-    var uri = utils.uriBase(req);
-    var resourceBaseUri = utils.filenameToBaseUri(
-        resourcePath,
-        uri,
-        root);
-
-    try {
-        $rdf.parse(
-            requestText,
-            resourceGraph,
-            resourceBaseUri,
-            contentType);
-    } catch (parseErr) {
-        debug("POST -- Error parsing resource: " + parseErr);
-        ldp.releaseResourceUri(resourcePath);
-        return next(new HttpError({
-            status: 400,
-            message: "Error parsing resource"
-        }));
-    }
-
-    // Add header link to the resource
-    header.addLinks(res, resourceMetadata);
-
-    // Finally, either create the new resource or container
-    if (resourceMetadata.isBasicContainer) {
-        resourcePath += '/';
-        resourceBaseUri += '/';
-        ldp.createNewContainer(
-            uri,
-            resourcePath,
-            resourceGraph,
-            containerCallback);
-    } else {
-        ldp.createNewResource(
-            req.hostname,
-            uri,
-            resourcePath,
-            resourceGraph,
-            resourceCallback);
-    }
-
-    function containerCallback(err) {
-        if (err) {
-            debug("POST -- Error creating new container: " + err.message);
-            return next(err);
-        }
-
-        debug("POST -- Created new container " + resourceBaseUri);
-        res.set('Location', resourceBaseUri);
-        res.sendStatus(201);
-        return next();
-    }
-
-    function resourceCallback(err) {
-        if (err) {
-            debug("POST -- Error creating resource: " + err.message);
-            return next(err);
-        }
-        res.set('Location', resourceBaseUri);
-        res.sendStatus(201);
-        return next();
-    }
+exports.handler = handler
+
+var debug = require('../debug').handlers
+var header = require('../header.js')
+var patch = require('./patch.js')
+var error = require('../http-error')
+
+function handler (req, res, next) {
+  var ldp = req.app.locals.ldp
+  var contentType = req.get('content-type')
+
+  // Handle SPARQL(-update?) query
+  if (contentType === 'application/sparql' ||
+      contentType === 'application/sparql-update') {
+    debug('POST -- Handling sparql query via PATCH')
+    return patch.handler(req, res, next)
+  }
+
+  var containerPath = req.path
+  debug('POST -- On parent: ' + containerPath)
+
+  // Not a container
+  if (containerPath[containerPath.length - 1] !== '/') {
```

Comment:
Do we currently have any other behavior for POST, except for creating new resources? In this case, shouldn't we just "correct" the URI by appending the missing `/`?